analyses/cell-type-scimilarity/scripts/02-run-scimilarity.py

```diff
@@ -10,9 +10,57 @@
 
 import anndata
 import pandas
+from scipy.sparse import csr_matrix
 from scimilarity import CellAnnotation
 from scimilarity.utils import align_dataset, lognorm_counts
 
+
+def format_scimilarity (adata: anndata.AnnData) -> anndata.AnnData:
+    """
+    Creates a new AnnData object formatted for running SCimilarity: 
+    
+    - Gene symbols, taken from the 'gene_symbol' column in 'adata.var' are used as var_names
+    If any duplicate gene symbols are found, they are collapsed by summing counts.
+    - The summed counts matrix is then stored in the 'counts' layer of the new AnnData object.
+
+    Parameters
+    ----------
+    adata : AnnData
+        Input AnnData object, with "gene_symbol" column in `adata.var`.
+
+    Returns
+    -------
+    AnnData
+        New AnnData object with gene symbols as var_names and counts stored in the 'counts' layer.
+    """
+
+    # Check that gene symbol column exists 
+    if not "gene_symbol" in adata.var.columns:
+        raise ValueError(
+            "The input AnnData object must have a 'gene_symbol' column in `adata.var`."
+        )
+    
+    # set gene symbols as var_names and make sure X has the raw counts
+    adata.var_names = adata.var["gene_symbol"].astype(str)
+    adata.X = adata.raw.X
+
+    # create a DataFrame with raw counts, dropping anything that doesn't have a gene symbol 
+    counts_df = adata.to_df().drop(columns=['nan'])
```

Comment:
I think you can keep this sparse if youare willing to be a bit more verbose.
```suggestion
    counts_df = pandas.DataFrame.sparse.from_spmatrix(
        adata.raw.X, 
        columns=adata.var["gene_symbol"].astype(str)
    ).drop(columns=['nan'])
```

Note that this should allow you to remove the two `adata` modifications as well (though I didn't start there, so didn't include that in the suggestion).